### PR TITLE
Log metrics on wandb

### PR DESCRIPTION
**Pull request by MatteoKartoon**
- Metrics logs added to wandb
- Function `evaluator` modified to take as parameters both tensor and image paths (file `metrics.py`, useful to compute metrics during the training)
- Visualization modified to display all the images in the test set
- Assertion bug fixed in train.py `assert config.finetune_last_epochs <= 0, "finetune_last_epochs must be less than 0"`
- Added file `fine_tune_hyperparameters.sh` to run experiments in a row, setting there some hyperparameters
- Learning rate decay implemented

### CODE REVIEW

File: scripts/train.py [hunk 1]
```diff
@@ -220,21 +261,26 @@ def _get_loss_key(self,epoch):
     def iteration_over_batches_validation(self, epoch, info_progress=None, step_idx=None, training_result=None):
         #Loop over the batches
         loss_components_dict={k:0 for k in self.loss_components_train.keys()}
+        validation_metrics_accumulated = {k:0 for k in config.display_eval_metrics}
```
Comment (joelkartoon) [inline scripts/train.py:264]: nit: keep naming consistent with `loss_components_dict`
👉 `validation_metrics_dict`

File: birefnet/config.py [hunk 2]
```diff
@@ -1,40 +1,46 @@
 import os
 import math
 
+#Set the default values for the parameters
+DEFAULT_LR = 1e-5
+DEFAULT_RUN_NAME = 'lr decay'
```
Comment (joelkartoon) [inline birefnet/config.py:6]: make default run name be based on date for ex

```
datetime.now().strftime("%Y%m%d_%H%M%S")
```

**Review by joelkartoon — APPROVED** [review]
